### PR TITLE
WarpX Static Attributes: Clean Up Types

### PR DESCRIPTION
**Pull request by EZoni**
I noticed these types while adding some Doxygen strings, and it seems to me like we could replace some of the `long` (or `int`) with `short` (based on https://github.com/ECP-WarpX/WarpX/pull/2916#discussion_r818253511) and some of the `int` with `bool` (I think seeing `bool` makes it easier for a developer to understand the nature of the variable, namely that it's a ON/OFF switch). Open to suggestions, though, this is just an idea.

### CODE REVIEW

File: Source/WarpX.cpp [hunk 1]
```diff
@@ -108,20 +108,20 @@ Real WarpX::quantum_xi_c2 = PhysConst::xi_c2;
...
-int WarpX::maxwell_solver_id;
-long WarpX::load_balance_costs_update_algo;
+short WarpX::current_deposition_algo;
+short WarpX::charge_deposition_algo;
+short WarpX::field_gathering_algo;
+short WarpX::particle_pusher_algo;
+short WarpX::maxwell_solver_id;
+short WarpX::load_balance_costs_update_algo;
 bool WarpX::do_dive_cleaning = 0;
 bool WarpX::do_divb_cleaning = 0;
```
Comment (ax3l) [inline Source/WarpX.cpp]: I would also replace those with `true`/`false` values and tell other developers about it in the development meeting.

Writing `0`/`1` is C style and confusing to new developers with a non-C background.

We should also talk in the dev meeting about extending the parser to support named boolean values, similar to CMake `if`:
https://cmake.org/cmake/help/latest/command/if.html#basic-expressions

basically case-insensitive support for True/on and False/off.

**Review by ax3l — APPROVED** [review]

File: Source/WarpX.cpp (hunk 1 above)
Comment (EZoni) [reply Source/WarpX.cpp]: Thank you, Axel. I replaced those and the corresponding PML flags as well in https://github.com/ECP-WarpX/WarpX/pull/2922/commits/6d26570b1bd261822f07cd84856658ce03fafccb.

**Review by ax3l — APPROVED** [review]
LGTM, thanks a lot!

**Comment by ax3l** [comment]
Technically, the algorithm choices should all be named `enum struct` or `struct { enum {}; };` types, on the long run.

**Pull request merged**